### PR TITLE
Fix cabal executable and test sections and fix warnings

### PR DESCRIPTION
**Pull request by hakujin**
Corrects project structure and .cabal file so that `cabal build` and
`cabal test` work. Refactors Main to reduce duplication and remove
`head` partial function. Enables `-Wall -Werror` and fixes warnings.

### CODE REVIEW

File: cal-haskell.cabal [hunk 1]
```diff
@@ -19,7 +19,7 @@ synopsis:            Trying to learn Haskell for great good!
 -- description:
 
 -- URL for the project homepage or repository.
-homepage:            git@github.com:erlandsona/cal_haskell.git
+homepage:            https://github.com/erlandsona/cal_haskell
```
Comment (erlandsona) [inline cal-haskell.cabal]: Just curious why https vs ssl?

File: src/Main.hs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import System.Environment (getArgs)
+
+import Month (monthString)
+import Year (yearString)
+
+main :: IO ()
+main = do
+  args <- map read <$> getArgs -- TODO: read will explode on non-numeric input
```
Comment (erlandsona) [inline src/Main.hs]: Why do you need to map read over/into an fmap of getArgs?
My guess is fmap lifts the values from get args into a regular list that can then just be mapped over?

File: test/DaySpec.hs [hunk 3]
```diff
@@ -20,7 +30,7 @@ spec = parallel $ describe "Day" $ do
       it "returns 5" $ zellers 1  1  2000 `shouldBe` 5
       it "returns 6" $ zellers 28 12 2999 `shouldBe` 6
 
-    context "Sad Path" $ do
+    context "Sad Path" $
```
Comment (erlandsona) [inline test/DaySpec.hs]: So this one really tripped me up... From what I've read do is the syntax for whenever you're dealing with IO() but when I was writing these tests some of them wouldn't work without it. But they all seemed to work with it.    why?

File: test/Spec.hs [hunk 4]
```diff
@@ -0,0 +1 @@
+{-# OPTIONS_GHC -F -pgmF hspec-discover #-}
```
Comment (erlandsona) [inline test/Spec.hs]: This I'm assuming is to be able to run cabal test?

File: cal-haskell.cabal (hunk 1 above)
Comment (hakujin) [inline cal-haskell.cabal]: I added a separate `source-repository` section below that covers the git specifics. This field is typically used for a HTTP page.

File: src/Main.hs (hunk 2 above)
Comment (hakujin) [inline src/Main.hs]: The types:
```haskell
getArgs :: IO [String]
map read :: Read a => [String] -> [a]
(<$>) :: Functor f => (a -> b) -> f a -> f b -- infix version of fmap
```

If we specialize `Read` to `Int` and `Functor` to `IO` and then substitute, we get:
```haskell
([String] -> [Int]) -> IO [String] -> IO [Int]
--  map read ^         getArgs ^      result ^
```

Read up on the `Functor` chapter in the Haskell Book for more info.

File: test/Spec.hs (hunk 4 above)
Comment (hakujin) [inline test/Spec.hs]: Check out the [hspec docs](http://hspec.github.io/hspec-discover.html) for more info.

File: test/DaySpec.hs (hunk 3 above)
Comment (hakujin) [inline test/DaySpec.hs]: Haskell uses `do` as [syntactic sugar](https://en.wikipedia.org/wiki/Syntactic_sugar) for Monadic bind (`>>=`). You'll probably want to check out the Monad chapter in the Haskell Book for a detailed explanation. [Here](https://en.wikibooks.org/wiki/Haskell/do_notation) are [two](http://www.haskellforall.com/2014/10/how-to-desugar-haskell-code.html) other resources.

**Comment by erlandsona** [comment]
Thanks a ton for all the help @hakujin! I read through the articles you mentioned... Learning more and more everyday... 

My buddy at Metova was telling me maybe a next useful program would be to write a plugin for neovim to handle window splits in a little more user friendly way... I've always wished Vim would just do a simple 4 way split EG: Like an X/Y coordinate plane. Window Management in Vim has always been such a royal pain... What are you're thoughts?

**Pull request merged**